multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ReferenceFinder.java

```diff
@@ -0,0 +1,157 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.cloudfoundry.multiapps.common.util.MiscUtil;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Hook;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.model.ProvidedDependency;
+import org.cloudfoundry.multiapps.mta.model.RequiredDependency;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.cloudfoundry.multiapps.mta.resolvers.ReferencePattern;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import jakarta.inject.Named;
+
+@Named("referenceFinder")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class ReferenceFinder {
+
+    private String parameterToFind;
+
+    private static final List<Pattern> COMPILED_PATTERNS = Arrays.asList(Pattern.compile(ReferencePattern.PLACEHOLDER.getPattern()),
```

Comment:
consider changing ReferencePattern to just have the compiled patterns saved and get those directly. Besides that - no need to persist another copy of the enum values here - just use the directly ReferencePattern.values()